pkg/resources/cloudconfig/openstack/cloudconfig.go

```diff
@@ -66,16 +66,39 @@ func ForCluster(cluster *kubermaticv1.Cluster, dc *kubermaticv1.Datacenter, cred
 	}
 
 	var lbClassOpts LBClassOpts
+	var lbClasses []kubermaticv1.LoadBalancerClass
+
 	if len(dc.Spec.Openstack.LoadBalancerClasses) > 0 {
 		lbClassOpts = make(LBClassOpts, len(dc.Spec.Openstack.LoadBalancerClasses))
-		for _, lbClass := range dc.Spec.Openstack.LoadBalancerClasses {
+		lbClasses = dc.Spec.Openstack.LoadBalancerClasses
+		for _, lbClass := range lbClasses {
+			lbClassOpts[lbClass.Name] = &LBClass{
+				FloatingNetworkID:  lbClass.Config.FloatingNetworkID,
+				FloatingSubnetID:   lbClass.Config.FloatingSubnetID,
+				FloatingSubnet:     lbClass.Config.FloatingSubnet,
+				FloatingSubnetTags: lbClass.Config.FloatingSubnetTags,
+				NetworkID:          lbClass.Config.NetworkID,
+				SubnetID:           lbClass.Config.SubnetID,
+				MemberSubnetID:     lbClass.Config.MemberSubnetID,
+			}
+		}
+	} else {
+		// initialize the map so we can add cluster-only classes below
+		lbClassOpts = make(LBClassOpts)
+	}
```

Comment:
That's good catch; thanks!
Since we are initializing the `lbClassOpts` variable regardless of the `dc.Spec.Openstack.LoadBalancerClasses` configuration, let's move the initialization of the LBClassOpts above, next to `lbClassOpts` declaration, as follows:

```go
	lbClassOpts := make(LBClassOpts)
```